### PR TITLE
Add re2 library

### PR DESCRIPTION
**Pull request by supechicken666**
Tested on `x86_64`

### CODE REVIEW

File: packages/re2.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require 'package'
+
+class Re2 < Package
+  description 'RE2 is a fast, safe, thread-friendly alternative to backtracking regular expression engines 
+like those used in PCRE, Perl, and Python.'
```
Comment (uberhacker) [inline packages/re2.rb:5]: Can we make this one line?  I think the wrapping may cause a problem.

File: packages/re2.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  source_url "https://github.com/google/re2/archive/#{@_ver}.tar.gz"
+  source_sha256 '1396ab50c06c1a8885fb68bf49a5ecfd989163015fd96699a180d6414937f33f'
+
+  def self.build
+    # cmake version doesn't provide libre2 library, use make instead
+    system 'make'
+  end
+  
+  def self.install
+    system "make DESTDIR=#{CREW_DEST_DIR} #{CREW_OPTIONS.gsub('--', '')} install"
```
Comment (uberhacker) [inline packages/re2.rb:21]: Are the compiler options lowercase?

Comment (supechicken) [reply packages/re2.rb:19]: (Yeah)[https://www.gnu.org/software/make/manual/html_node/Directory-Variables.html#Directory-Variables]

File: packages/re2.rb (hunk 1 above)
Comment (supechicken) [reply packages/re2.rb]: Removed

**Pull request merged**